### PR TITLE
 [FLINK-12515][Build System, BuildSystem / Shaded] Sync flink-shaded-hadoop with dependency management entries

### PR DESCRIPTION
**Pull request by sunjincheng121**
In this PR will add ` dependency management` entries for `flink-shaded-hadoop`. 

Process of consideration:
 1. DependencyList -> list all the dependency for `flink-shaded-hadoop` by `mvn dependency:tree`
 2.  TargetDependencyList -> find out the `artifactId` both in `DependencyList` and `flink-parent's dependencyManagement`. 
 3. add the `TargetDependencyList` into ` dependency management` of `flink-shaded-hadoop-2` in `flink-shaded`.

### CODE REVIEW

**Comment by sunjincheng121** [comment]
Hi, @zentol I am grateful if you have time to review the changes!
Best,Jincheng

**Comment by zentol** [comment]
`flink-shaded-hadoop-2-uber` bundles additional dependencies, is it really unaffected by the dependency management entries?

**Comment by zentol** [comment]
ah, `flink-shaded-hadoop-2-uber`picks up on the dependency management entries in here, that's convenient...

File: flink-shaded-hadoop-2/pom.xml [hunk 1]
```diff
@@ -36,37 +36,202 @@ under the License.
...
+				<version>4.5.3</version>
+			</dependency>
+			<dependency>
+				<groupId>org.apache.httpcomponents</groupId>
+				<artifactId>httpcore</artifactId>
+				<version>4.4.6</version>
+			</dependency>
+			<dependency>
+				<groupId>org.apache.commons</groupId>
+				<artifactId>commons-lang3</artifactId>
```
Comment (zentol) [inline flink-shaded-hadoop-2/pom.xml]: I think this isn't bundled anywhere; we could drop the entry in that case. (default, check other hadoop version)

File: flink-shaded-hadoop-2/pom.xml [hunk 2]
```diff
@@ -36,37 +36,202 @@ under the License.
...
+				<version>1.3.9</version>
+			</dependency>
+			<dependency>
+				<groupId>org.slf4j</groupId>
+				<artifactId>slf4j-api</artifactId>
+				<version>${slf4j.version}</version>
+			</dependency>
+			<dependency>
+				<groupId>junit</groupId>
+				<artifactId>junit</artifactId>
```
Comment (zentol) [inline flink-shaded-hadoop-2/pom.xml]: we should be able to ignore test dependencies.

**Comment by sunjincheng121** [comment]
That' correct, `flink-shaded-hadoop2` is the only dependency of `flink-shaded-hadoop-2-uber` , and `flink-shaded-hadoop2`'s dependency management will also get benefit in `flink-shaded-hadoop-2-uber`.

**Comment by sunjincheng121** [comment]
> I think we can reduce the entries a bit. Test dependencies are irrelevant for packaging, so these can be removed. (hamcrest, junit, byte-buddy etc)
> 
> I did find 2 differences in `flink-shaded-hadoop-2-uber` for all hadoop versions: It now bundles jline and netty 3.10.5.Final. This shouldn't be the case; while we do have a dependency management entry in `flink-shaded-hadoop-2`, there are IIRC some strange rules as to when dependency management entries are actually applied; if module A has such an entry, and module B depends on A, then B is not necessarily applying them as well.
> 
> In short, you have to duplicate (or move) the zookeeper exclusions into `flink-shaded-hadoop-2-uber`

Sorry, do you meant the following section? 
```
<dependency>
				<groupId>org.apache.zookeeper</groupId>
				<artifactId>zookeeper</artifactId>
				<version>${zookeeper.version}</version>
				<exclusions>
					<exclusion>
						<groupId>log4j</groupId>
						<artifactId>log4j</artifactId>
					</exclusion>
					<exclusion>
						<groupId>org.slf4j</groupId>
						<artifactId>slf4j-log4j12</artifactId>
					</exclusion>
					<!-- Netty is only needed for ZK servers, not clients -->
					<exclusion>
						<groupId>io.netty</groupId>
						<artifactId>netty</artifactId>
					</exclusion>
					<!-- jline is optional for ZK console shell -->
					<exclusion>
						<groupId>jline</groupId>
						<artifactId>jline</artifactId>
					</exclusion>
				</exclusions>
			</dependency>
```
Dou you mean that you find that the `netty` and `jline` have packaged into `flink-shaded-hadoop-2-uber-2.4.1-7.0.jar` ?

**Comment by sunjincheng121** [comment]
Thanks for your review @zentol 

I have removed the `test` and `provided` dependency.
Please have another look, feel free to tell me if there anything I miss.

Regards,
Jincheng

**Pull request merged**